leshan-core/src/main/java/org/eclipse/leshan/core/request/ObserveCompositeRequest.java

```diff
@@ -88,13 +88,31 @@ public ObserveCompositeRequest(ContentFormat requestContentFormat, ContentFormat
             List<LwM2mPath> paths, Object coapRequest) {
         super(coapRequest);
 
+        validatePaths(paths);
         this.requestContentFormat = requestContentFormat;
         this.responseContentFormat = responseContentFormat;
         this.paths = paths;
 
         this.context = Collections.emptyMap();
     }
 
+    private void validatePaths(List<LwM2mPath> paths) throws InvalidRequestException {
+        if (paths == null || paths.size() == 0)
+            throw new InvalidRequestException("Path is mandatory");
+
+        // Ensure there is no overlapped Path (e.g. "3/0" and "/3/0/1")
+        for (int i = 0; i < paths.size(); i++) {
+            LwM2mPath firstPath = paths.get(i);
+            for (int j = i + 1; j < paths.size(); j++) {
+                LwM2mPath secondPath = paths.get(j);
+                if (firstPath.startWith(secondPath) || secondPath.startWith(firstPath)) {
+                    throw new InvalidRequestException("Invalid path list :  %s and %s are overlapped paths", firstPath,
+                            secondPath);
+                }
+            }
+        }
```

Comment:
Maybe we should factorize that code ? 
I don't know where to put it :thinking:

Maybe we add a static method in LwM2mPath like : 
```java
public static boolean containOverlappedPath(List<LwM2mPath> paths)
```

Reply:
I added something of the like, I think making it void and throw an exception is a better idea - what do you think?

Reply:
I'm not sure that `LwM2mPath` class should  raise `InvalidRequestException`, I think this is maybe not so reusable. :thinking:

Reply:
You're right, though I still don't like the validation method returning boolean - then if it fails we don't have error message and we have to duplicate it every single time, so that kind of defeats the purpose. Maybe we can do something like:

```java
public static void validateNotOverlapping(List<LwM2mPath> paths) throws RuntimeException {
    ...
}
public static void validateNotEmpty(List<LwM2mPath> paths) throws RuntimeException {
    ...
}

...

try {
    LwM2mPath.validateNotEmpty(paths);
    LwM2mPath.validateNotOverlapping(paths);
} catch(RuntimeException exception) {
    throw new InvalidRequestException(exception.getMessage());
}
```

Reply:
Yep you're right my with my way we lost the error message.
You can see in `LwM2mNodeUtil`, there is way to solve this return a String errorMessage instead of boolean.
(I don't know if that sounds good to you?)

You're way works too but raising RuntimeException sounds strange.
I'm not sure `InvalidLwM2mPathException` is adapted as path are not really invalid ?
or IllegalArgumentException like in `Validate` class ?

Reply:
With String return it would be:
```java
public static String validateNotOverlapping(List<LwM2mPath> paths) {
    //returns error message string or null if everything is good
}
public static String validateNotEmpty(List<LwM2mPath> paths) {
    //same as above
}

...

String validateNotEmptyMessage = LwM2mPath.validateNotEmpty(paths);
if(validateNotEmptyMessage != null) {
    throw new InvalidRequestException(validateNotEmptyMessage);
}
String validateNotOverlappingMessage = LwM2mPath.validateNotOverlapping(paths);
if(validateNotOverlappingMessage != null) {
    throw new InvalidRequestException(validateNotOverlappingMessage);
}
```

So it's pretty long and awkward IMO. We could also just use `IllegalArgumentException` or `InvalidLwM2mPathException` instead of `RuntimeException` in my last concept - that's up to you how you want it.

Reply:
Ok let's go for throwing an exception probably IllegalArgumentException is better.
Note that org.eclipse.leshan.core.util.Validate.notEmpty(Collection) already to this kind of check

Reply:
OK, I changed the exception class and removed empty check from `LwM2mPath` and used the one from `Validate`. Hope that's finally good.